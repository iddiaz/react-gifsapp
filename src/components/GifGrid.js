// import React, {useState, useEffect } from 'react'
import React from 'react'
import { GifGridItem } from './GifGridItem';

import { useFetchGifs } from './../hooks/useFetchGifs';

export const GifGrid = ({category}) => {


   const {data:images , loading } = useFetchGifs( category );




   return (
      <>
         <h3> {category} </h3>
         <div className="card-grid">

            { loading && <p className="animate__animated animate__flash">Loading</p>  }
            
               {
                  images.map(( img) => (
                     <GifGridItem 
                        key={img.id} 
                        // img={img} 
                        {...img}
                        />
                  ))
               }            
         </div>      
      </>
   )
}
