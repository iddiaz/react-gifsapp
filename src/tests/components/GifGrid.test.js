import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import React from 'react';
import '@testing-library/jest-dom';
import { useFetchGifs } from './../../hooks/useFetchGifs';
jest.mock('./../../hooks/useFetchGifs');

describe('Pruebas en <GifGrid/>', ()=>{

   const category = 'One punch';
   
   
   test('deberia cargar correctamente', ()=> {
      
      useFetchGifs.mockReturnValue({
         data: [],
         loading: true
      })
   
      const wrapper = shallow( <GifGrid category={category}/>) 
      expect( wrapper ).toMatchSnapshot();

   })
   test('deberia mostrar items cuando se cagan imÃ¡genes pruebas de useFetchGifs', ()=> { 

      const gifs = [{
         id: 'ABC',
         url: 'https://localhost/cualquier/lugar.jpg',
         title: 'un titulo cualquiera'
      }];

      useFetchGifs.mockReturnValue({
         data: gifs,
         loading: false
      });
      
      const wrapper = shallow( <GifGrid category={category}/>)
      
      // expect( wrapper ).toMatchSnapshot();
      expect( wrapper.find('p').exists()).toBe(false);
      expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );

      
      
   }) 

})