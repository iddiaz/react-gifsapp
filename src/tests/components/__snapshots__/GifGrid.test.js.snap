// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid/> deberia cargar correctamente 1`] = `
<Fragment>
  <h3>
     
    One punch
     
  </h3>
  <div
    className="card-grid"
  >
    <p
      className="animate__animated animate__flash"
    >
      Loading
    </p>
  </div>
</Fragment>
`;
